coreos.sh

```diff
@@ -59,9 +62,9 @@ extract_image() {
        ;;
       *)return 1;;
     esac
-
+    #TODO: what happens if COMPRESSION is empty because $2 is "bin"?
     # extract image with given compression
-    $COMPRESSION -d --stdout $EXTRACTFROM > ${DRIVE1}; EXITCODE=$?
+    "$COMPRESSION -d --stdout $EXTRACTFROM" > "${DRIVE1}"; EXITCODE=$?
```

Comment:
Am 21.01.2016 um 20:02 schrieb taxus13:
> In coreos.sh
> <https://github.com/virtapi/installimage/pull/22#discussion_r50445985>:
>
> >      # extract image with given compression
> > -    $COMPRESSION -d --stdout $EXTRACTFROM > ${DRIVE1}; EXITCODE=$?
> > +    "$COMPRESSION -d --stdout $EXTRACTFROM" > "${DRIVE1}"; EXITCODE=$?
>
> is it neccessary to put the whole command in quotemarks?
This will give a nice "command not found", because there are no commands with spaces in there. A »"$COMPRESSION" -d --stdout "$EXTRACTFROM" > "$DRIVE1"; declare -i EXITCODE="$?"« would be a proper way.